File: InterviewQuestions/Python Interview Programming Questions/task14.py
# Write a Python program to add 20 unique four digit numbers to an array

import random

number_array = []

while len(number_array) < 20:

    random_number = str(random.randint(1000, 9999))

    print(random_number)

    if len(random_number) == 4:

        match_found = False

        for a in number_array:

            if str(a) == random_number:

                match_found = True

        if not match_found:

            number_array.append(int(random_number))
            
print(number_array)

File: InterviewQuestions/Javascript Interview Programming Questions/task14.js
// Write a Javascript program to add 20 unique four digit numbers to an array

let number_array = []

while (number_array.length < 20) {

    random_number = String(Math.floor(Math.random() * 10000))
    console.log(random_number);

    if (random_number.length === 4) {

        let match_found = false
        
        for (let a of number_array) {

            if (String(a) === random_number) {

                match_found = true
            }
        }

        if (match_found === false) {

            number_array.push(Number(random_number))
        }
    }
}

console.log(number_array)